samcli/commands/local/invoke/cli.py

```diff
@@ -26,40 +26,47 @@
 Invoking a Lambda function using input from stdin
 $ echo '{"message": "Hey, are you there?" }' | sam local invoke "HelloWorldFunction" \n
 """
+STDIN_FILE_NAME = "-"
 
 
 @click.command("invoke", help=HELP_TEXT, short_help="Invokes a local Lambda function once.")
 @click.option("--event", '-e',
               type=click.Path(),
-              default="-",  # Defaults to stdin
+              default=STDIN_FILE_NAME,  # Defaults to stdin
               help="JSON file containing event data passed to the Lambda function during invoke. If this option "
                    "is not specified, we will default to reading JSON from stdin")
+@click.option("--no-args", is_flag=True, default=False, help="Invoke Function without parameters")
```

Comment:
Thinking about this more, I don't think `--no-args` describes what this does. This is really giving a default value to the event. Maybe something like `--no-event` (so it matches with the rest of the command) and then update the help text to be `Invoke the Function with an empty event dictionary`?

@SoodDhruv Do you have any thoughts here?

Reply:
How about not requiring the --no-args/--no-event option? 'sam local invoke "FunctionName"' could just invoke the function with a default payload.

Reply:
@SoodDhruv `sam local invoke "FunctionName"` currently allows the user to use STDIN to provide the event payload. This would cause a conflict with current behavior. What do you think?
@jfuss I agree. I like `--no-event` better for the option name.

Reply:
@akkaash That makes sense. I can see an argument for changing this behavior but something maybe we shouldn't do now. Let's just leave this as `--no-event` and then we can see if there is a path/reason to change this behavior to what @SoodDhruv suggested.